# HG changeset patch
# User Rik <rik@octave.org>
# Date 1382470051 25200
#      Tue Oct 22 12:27:31 2013 -0700
# Node ID 36d646ead233d1bc91c42617939e07bf0dc60471
# Parent  d449f4668b721f595d321de0626bf6cb8ce0010a
Fix segfault when x or y data values are large (bug #39925).

* libinterp/corefcn/gl-render.cc(draw_image): Check result of transform
calculation.  If there are any NaN values, calc has failed and abort
drawing image.

diff --git a/libinterp/corefcn/gl-render.cc b/libinterp/corefcn/gl-render.cc
--- a/libinterp/corefcn/gl-render.cc
+++ b/libinterp/corefcn/gl-render.cc
@@ -2552,16 +2552,22 @@ opengl_renderer::draw_image (const image
     x(1) = x(1) + (w-1);
 
   if (h > 1 && y(1) == y(0))
     y(1) = y(1) + (h-1);
 
   const ColumnVector p0 = xform.transform (x(0), y(0), 0);
   const ColumnVector p1 = xform.transform (x(1), y(1), 0);
 
+  if (xisnan (p0(0)) || xisnan (p0(1)) || xisnan (p1(0)) || xisnan (p1(1)))
+    {
+      warning ("gl-render: image x,y data too large to draw");
+      return;
+    }
+
   // image pixel size in screen pixel units
   float pix_dx, pix_dy;
   // image pixel size in normalized units
   float nor_dx, nor_dy;
 
   if (w > 1)
     {
       pix_dx = (p1(0) - p0(0))/(w-1);
